eos-tech-support/eos-download-image

```diff
@@ -79,8 +79,13 @@ class EosDownloadImage(object):
         headers['Accept-Encoding'] = 'identity'
 
         # Make a HEAD request to get the size and mtime
-        resp = self.session.head(src, headers=headers, timeout=TIMEOUT)
+        resp = self.session.head(src, headers=headers, timeout=TIMEOUT,
+                                 allow_redirects=True)
         if resp.status_code == 200:
+            if src != resp.url:
+                log("Followed redirect to", resp.url)
+                src = resp.url
+
             src_mtime = dateutil.parser.parse(resp.headers['Last-Modified'])
             src_size = int(resp.headers['Content-Length'])
         elif resp.status_code == 403:
```

Comment:
It occurs to me that by pointing to a server we control, rather than directly to CloudFront, we could actually fix this problem where HEAD is not supported, by getting the Content-Length from s3fs.

Reply:
I'm not sure where to keep track of that idea, but it's a good one!